service/history/replication/ack_manager.go

```diff
@@ -686,3 +686,25 @@ func getVersionHistoryItems(
 	}
 	return versionhistory.CopyVersionHistory(versionHistoryBranch).GetItems(), versionHistoryBranch.GetBranchToken(), nil
 }
+
+func (p *ackMgrImpl) handleReadHistoryError(
+	namespaceID string,
+	workflowID string,
+	runID string,
+	err error,
+) error {
+	switch err.(type) {
+	case *serviceerror.NotFound, *serviceerror.DataLoss:
```

Comment:
Why `NotFound` is here too?

Reply:
If the first batch of the history is lost, it will return not found. https://github.com/temporalio/temporal/blob/15c52675e7633ffc3a043956d5e18380a0fbaefd/common/persistence/history_manager.go#L987

Reply:
One assumption here is history will be there as long as mutable state is here.